packages/@aws-cdk/aws-ec2/lib/instance.ts

```diff
@@ -0,0 +1,263 @@
+
+  /**
+   * The length of time to wait for the resourceSignalCount
+   *
+   * The maximum value is 43200 (12 hours).
+   *
+   * @default Duration.minutes(5)
+   */
+  readonly resourceSignalTimeout?: Duration;
+
+  /**
+   * VPC to launch the instance in.
+   */
+  readonly vpc: IVpc;
+
+  /**
+   * Type of instance to launch
+   */
+  readonly instanceType: InstanceType;
+
+  /**
+   * AMI to launch
+   */
+  readonly machineImage: IMachineImage;
+
+  /**
+   * Specific UserData to use
+   *
+   * The UserData may still be mutated after creation.
+   *
+   * @default - A UserData object appropriate for the MachineImage's
+   * Operating System is created.
+   */
+  readonly userData?: UserData;
+
+  /**
+   * An IAM role to associate with the instance profile assigned to this Auto Scaling Group.
+   *
+   * The role must be assumable by the service principal `ec2.amazonaws.com`:
+   *
+   * @example
+   *
+   *    const role = new iam.Role(this, 'MyRole', {
+   *      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com')
+   *    });
+   *
+   * @default A role will automatically be created, it can be accessed via the `role` property
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * The name of the instance
+   *
+   * @default - CDK generated name
+   */
+  readonly instanceName?: string;
+
+}
+
+export class Instance extends Resource implements IConnectable {
+
+  /**
+   * The type of OS the instance is running.
+   */
+  public readonly osType: OperatingSystemType;
+
+  /**
+   * Allows specify security group connections for the instance.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * The IAM role assumed by the instance.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * UserData for the instance
+   */
+  public readonly userData: UserData;
+
+  /**
+   * the underlying instance resource
+   * @attribute
+   */
+  public readonly instance: CfnInstance;
+  /**
+   * @attribute
+   */
+  public readonly instanceId: string;
+  /**
+   * @attribute
+   */
+  public readonly instanceAvailabilityZone: string;
+  /**
+   * @attribute
+   */
+  public readonly instancePrivateDnsName: string;
+  /**
+   * @attribute
+   */
+  public readonly instancePrivateIp: string;
+  /**
+   * @attribute
+   */
+  public readonly instancePublicDnsName: string;
+  /**
+   * @attribute
+   */
+  public readonly instancePublicIp: string;
+
+  private readonly securityGroup: ISecurityGroup;
+  private readonly securityGroups: ISecurityGroup[] = [];
+
+  constructor(scope: Construct, id: string, props: InstanceProps) {
+    super(scope, id);
+
+    this.securityGroup = new SecurityGroup(this, 'InstanceSecurityGroup', {
+      vpc: props.vpc,
+      allowAllOutbound: props.allowAllOutbound !== false
+    });
+    this.connections = new Connections({ securityGroups: [this.securityGroup] });
+    this.securityGroups.push(this.securityGroup);
+    this.node.applyAspect(new Tag(NAME_TAG, props.instanceName || this.node.path));
```

Comment:
Use `Tag.add`

Reply:
I copied this from AutoScalingGroup but I changed it in this PR now.